### PR TITLE
Add API wrapper for "Plugins"

### PR DESCRIPTION
**Pull request by bhks**
## Description

While working with Grafana I found this amazing python client, I used it to explore the plugin Installation feature using python program. While Install/unInstall APIs are not explicitly called out in Grafana API document. I found it useful to have the base layer on top of which community can build examples and write their own version of how to manage plugins via APIs.

Plugin API link : https://github.com/grafana/grafana/blob/v9.4.x/pkg/api/api.go#L423 

Please provide feedback for improvement needed and I should be able to incorporate.

## Checklist

- [ ] The patch has appropriate test coverage
- [x] The patch follows the style guidelines of this project
- [x] The patch has appropriate comments, particularly in hard-to-understand areas
- [x] The documentation was updated corresponding to the patch
- [x] I have performed a self-review of this patch

### CODE REVIEW

**Comment by amotl** [comment]
Dear Bhagwat,

thank you for contributing this excellent patch, I love it, and I think it will be a good candidate to be leveraged by the downstream [grafana-wtf](https://github.com/panodata/grafana-wtf) package.

May I humbly ask you to take care about code syle / formatting, and add a corresponding software test case [^1]? You can find relevant instructions within [docs/development.md](https://github.com/panodata/grafana-client/blob/main/docs/development.md). Thanks!

With kind regards,
Andreas.

[^1]: Saying this, the test suite currently does not have that much value, but at least we have it, and we may migrate it to run real integration tests in the future, see GH-50.

**Comment by bhks** [comment]
> Dear Bhagwat,
> 
> thank you for contributing this excellent patch, I love it, and I think it will be a good candidate to be leveraged by the downstream [grafana-wtf](https://github.com/panodata/grafana-wtf) package.
> 
> May I humbly ask you to take care about code syle / formatting, and add a corresponding software test case [1](#user-content-fn-1-fbe3d691a44a9588df652d4f05bd8566)? You can find relevant instructions within [docs/development.md](https://github.com/panodata/grafana-client/blob/main/docs/development.md). When running `poe format`, which is missing to be outlined in the documentation, most of the code style errors will be fixed automatically.
> 
> Other than this, if you can afford the time, please also add an item to the list of supported features at [README » Overview](https://github.com/panodata/grafana-client#overview).
> 
> Thank you in advance, and with kind regards, Andreas.
> 
> ## Footnotes
> 1. Saying this, the test suite currently does not have that much value, but at least we have it, and we may migrate it to run real integration tests in the future, see [Improve test suite by adding integration tests with Grafana #50](https://github.com/panodata/grafana-client/issues/50). [↩](#user-content-fnref-1-fbe3d691a44a9588df652d4f05bd8566)

Absolutely thanks for calling out the `poe format` missing. I was trying to just run `poe lint` and that was giving me error.

> Add a corresponding software test case

I think its a good callout, but as you have mentioned we may not have any benefit on the unit test. I have some example code which I believe we can use to add as Integ test. You can find my changes here :https://github.com/panodata/grafana-client/compare/main...bhks:grafana-client:main#diff-dc47e20d0c4b3322afa473cbebd824e602265a21bc5b628bb4e2de5496d37a9a

I wanted to first see how my first PR goes and get an opportunity to learn what is the process and how we are thinking about this.

**Comment by bhks** [comment]
>Other than this, if you can afford the time, please also add an item to the list of supported features at [README » Overview](https://github.com/panodata/grafana-client#overview).

Updated the Readme, Thanks for pointing that out.

**Comment by bhks** [comment]
> When running poe format, which is missing to be outlined in the documentation, most of the code style errors will be fixed automatically.

Added to Development document to run `poe format`

File: README.md [hunk 1]
```diff
@@ -168,31 +168,32 @@ versions of Grafana might not support certain features or subsystems.
...
+| Folder/Dashboard Search        | +- |
+| Health                         | + |
+| Organisation                   | + |
+| Other                          | + |
+| Plugin                         | + |
+| Preferences                    | + |
+| Rbac                           | +- |
+| Snapshot                       | + |
+| Teams                          | + |
+| User                           | + |
```
Comment (amotl) [inline README.md:196]: It looks like this whole block got reformatted. I know it's a pain, my editor is also playing tricks on me here. Can I ask you to edit only the spot at hand? Apologies for the nit.

File: docs/development.md [hunk 2]
```diff
@@ -19,6 +19,9 @@ poe test
 python -m unittest -k preference -vvv
 ```
 
+### Formatting
+When Creating a PR you can run `poe format` which will help remove code style issues and keep the repo clean.
```
Comment (amotl) [inline docs/development.md:23]: Just a suggestion.
```suggestion
Before creating a PR, you can run `poe format`, in order to resolve code style issues.
```

**Review by amotl — CHANGES_REQUESTED** [review]
Thanks for bringing in those improvements. I've added two more nits.

**Comment by amotl** [comment]
## Adding integration tests

> I think its a good callout, but as you have mentioned we may not have any benefit on the unit test.

At least it will run and cover the added code, that is still valuable.

> I have some example code which I believe we can use to add as an integration test. You can find my changes at https://github.com/panodata/grafana-client/compare/main...bhks:grafana-client:main#diff-dc47e20d0c4b3322afa473cbebd824e602265a21bc5b628bb4e2de5496d37a9a
>
> I wanted to first see how my first PR goes and get an opportunity to learn what is the process and how we are thinking about this.

Adding a trimmed variant of your example programs to the `examples` folder will be well appreciated on behalf of another patch. Maybe, when not installing the whole shebang of what is shipped with `aws-grafana`, but using a trimmed-down version instead, for demonstration purposes, the total runtime (probably determined by download speed and -efficiency) can be reduced, so we can think about including it into corresponding integration tests.

For bringing in the test framework itself, I was planning to reuse existing code from Kotori and grafana-wtf.

- https://github.com/daq-tools/kotori/blob/main/test/conftest.py
- https://github.com/panodata/grafana-wtf/blob/main/tests/conftest.py

Thank you for your commitment, I appreciate it very much.

**Comment by bhks** [comment]
> At least it will run and cover the added code, that is still valuable.

Yeah let me find some time to add that test.

>The total runtime (probably determined by download speed and -efficiency) can be reduced, so we can think about including it into corresponding integration tests.

Based on my testing the download is pretty fast specially the Grafana Labs marketplace binaries, which are available here https://grafana.com/api/plugins 

with links like following: 

```
      "packages": {
        "linux-amd64": {
          "md5": "4e2e3e22a568688b3141067804d76e6d",
          "sha256": "98d32c61fbb8cda667cfe64a097b57a762a77d5a0c90458d78885af9c97d5b30",
          "packageName": "linux-amd64",
          "downloadUrl": "/api/plugins/alexanderzobnin-zabbix-app/versions/4.4.1/download?os=linux&arch=amd64"
        }
```

So it might not cause issues, when we run the install tests in parallel to the other tests.

**Comment by bhks** [comment]
Thanks for calling out on Unit tests, it still have values and helps set the right URLs and give us an opportunity to review our code carefully.

**Comment by amotl** [comment]
Perfect, thank you so much!

**Comment by amotl** [comment]
Dear Bhagwat. Your improvements have been included in release 3.8.0. Thank you again!

**Pull request merged**